lib/types/SSI.types.ts

```diff
@@ -54,16 +62,200 @@ export interface Issuer {
   [x: string]: unknown;
 }
 
-export interface Credential {
-  '@context': string[];
-  id: string;
-  type: string[];
+export interface InternalCredential {
+  getAudience(): string | undefined;
+
+  getBaseCredential(): InternalCredentialBase;
+
+  getContext(): string[] | string;
+
+  getExpirationDate(): string | undefined;
+
+  getId(): string | undefined;
+
+  getIssuer(): unknown;
```

Comment:
What is the reason for the unknown? Isn't this either a string or an Issuer object?